### PR TITLE
Introduced utility `HttpServletFilter`

### PR DESCRIPTION
**Pull request by jglick**
I have come across a number of uses of `PluginServletFilter.addFilter` over the years. While it works, it has some minor drawbacks:

* Since it is a procedural call, you must do a little extra work to run an `@Initializer`. Most plugin contributions are expressed as `@Extension`s (which are also orderable if the need arises).
* You get a `ServletRequest` and `ServletResponse` and have to cast, even though Jetty is never going to be handling any other kind of request.
* The `FilterChain` idiom is somewhat low-level. Most implementations wind up either delegating at the end, or not.
* When a lot of filters are installed, the stack of an HTTP handler thread can get pretty deep. For someone looking at a thread dump, most if not all of those methods are irrelevant because they are in a tail call.

This new utility makes for simpler usage in typical cases. Shown here with a couple of example usages in core. Usages from `SetupWizard` and `HudsonPrivateSecurityRealm` would not be good candidates since they involve dynamically inserting or retracting a filter. There are other cases where the chain delegation is not a tail call ([example](https://github.com/jenkinsci/timemachine-plugin/blob/d2f32fd9cf424e4a0605535df1c6d2e52c9bbd21/src/main/java/fr/loof/jenkins/timemachine/TimeMachineFilter.java#L32-L35)) which would likewise not be suited to the new API.

### Testing done

Copied from a utility used successfully in a proprietary plugin (with tests).

### Proposed changelog entries

- Added a utility `HttpServletFilter` to the API.

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7892"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
